conftest.py

```diff
@@ -29,46 +31,44 @@ def pytest_configure(config):
     reporter = FixtureReporter(config)
     config.pluginmanager.register(reporter, 'fixturereporter')
 
+def _max_fname_len(file_system_path, default_max_len=80):
+    name_max_str = [s for s in os.pathconf_names if s.endswith('_NAME_MAX')]
+    if len(name_max_str) == 1:
+        try:
+            return os.pathconf(file_system_path, name_max_str[0])
+        except OSError:
+            pass
+    return default_max_len
+
+def _make_fname_valid(file_system_path, fname, len_margin):
+    valid_fname_chars = set(string.ascii_letters+string.digits+'-.')
+    max_len = _max_fname_len(file_system_path)-len_margin
+    return ''.join((c if c in valid_fname_chars else '_') for c in fname[:max_len])
+
+@contextlib.contextmanager
+def _tmpdir_aux(request, tmpdir_factory, scope, name):
+    basetemp = str(tmpdir_factory.getbasetemp())
+    name = _make_fname_valid(file_system_path=basetemp, fname=name, len_margin=50)
+    tmpdir = tempfile.mkdtemp(dir=basetemp, prefix='test-{}-{}-'.format(scope, name))
+    yield tmpdir
+    if os.path.isdir(tmpdir): shutil.rmtree(tmpdir)
```

Comment:
Don't we still think we should `shutil.rmtree(ignore_errors=True)`? Our error before in #839 was not about the tmpdir not existing, it was about a file within the tmpdir disappearing while rmtree was in the middle of walking the tree.

And if we `ignore_errors=True` then the `if os.path.isdir` becomes unnecessary as a side effect.